crates/jstzd/tests/octez_node_test.rs

```diff
@@ -0,0 +1,63 @@
+use jstzd::task::{octez_node, Task};
+
+#[ignore = "ignore for now until octez binaries in CI are set up properly"]
+#[tokio::test(flavor = "multi_thread")]
+async fn octez_node_test() {
+    let data_dir = tempfile::tempdir().unwrap();
+    let log_file = tempfile::NamedTempFile::new().unwrap();
+    let port = std::net::TcpListener::bind("127.0.0.1:0")
+        .unwrap()
+        .local_addr()
+        .unwrap()
+        .port();
+    let rpc_endpoint = format!("localhost:{}", port);
+
+    let mut config_builer = octez_node::OctezNodeConfigBuilder::new();
+    config_builer
+        .set_binary_path("octez-node")
+        .set_data_dir(data_dir.path().to_str().unwrap())
+        .set_network("sandbox")
+        .set_rpc_endpoint(&rpc_endpoint)
+        .set_log_file(log_file.path().to_str().unwrap())
+        .set_run_options(&[]);
+    let mut f = octez_node::OctezNode::spawn(config_builer.build().unwrap())
+        .await
+        .unwrap();
+
+    let health_check_endpoint = format!("http://{}/health/ready", rpc_endpoint);
+    // Should be able to hit the endpoint since the node should have been launched
+    let mut node_ready = false;
+    for _ in 0..10 {
+        tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
+        let res = reqwest::get(&health_check_endpoint).await;
+        if let Ok(raw_body) = res {
+            let body = raw_body
+                .json::<std::collections::HashMap<String, bool>>()
+                .await
+                .unwrap();
+            if let Some(v) = body.get("ready") {
+                node_ready = *v;
+                if node_ready {
+                    break;
+                }
+            }
+        }
+    }
+    assert!(node_ready);
+
+    let _ = f.kill().await;
+    // Wait for the process to shutdown entirely
+    let mut node_destroyed = false;
+    for _ in 0..10 {
+        tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
+        let res = reqwest::get(&health_check_endpoint).await;
+        // Should get an error since the node should have been terminated
+        if let Err(e) = res {
+            node_destroyed = e.to_string().contains("Connection refused");
+            if node_destroyed {
+                break;
+            }
+        }
+    }
```

Comment:
The polling can be factored out - DRY

Reply:
This will be replaced by the health check in #566.

Reply:
Ah yes, sure. Then let's just factor out the retries? 
`pub fn retry(f: Fn() -> Option<T>, retries: u16, interval_ms: u32) -> Result<T> `

Reply:
Sorry I forgot that the retry part will not be replaced. I don't think this is that helpful here at this moment, but I've moved the retry logic out anyway.